SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -539,10 +541,19 @@ void ActorSystem::HandleActorAuthority(const Worker_EntityId EntityId, const Wor
 						Actor->RemoteRole = ROLE_AutonomousProxy;
 
 						// Flush PC interest on handover
-						if (GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries
-							&& GetDefault<USpatialGDKSettings>()->bRefreshClientInterestOnHandover)
+						if (GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries)
 						{
-							Channel->MarkInterestDirty();
+							Worker_EntityId ControllerEntityId =
+								NetDriver->PackageMap->GetEntityIdFromObject(Actor->GetNetConnection()->PlayerController);
+							if (ControllerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+							{
+								MarkClientInterestDirty(ControllerEntityId, /*bOVerwrite*/ true);
+							}
+							else
+							{
+								UE_LOG(LogActorSystem, Warning, TEXT("Failed to get player controller to update client interest (%s)"),
```

Comment:
Receiving spatialos network data with no known entity ID seems error-worthy?
```suggestion
								UE_LOG(LogActorSystem, Error, TEXT("Failed to get player controller to update client interest (%s)"),
```

Reply:
It's getting the entity id of the entities owning controller. There could be a race where the owned object comes down first, and the controller second.

Reply:
Worth adding as a comment I think.

Reply:
> There could be a race where the owned object comes down first, and the controller second.
 
Client interest is only defined for PCs. Why don't we call `MarkClientInterestDirty` when **only** the PC is received?

Reply:
> Client interest is only defined for PCs. Why don't we call `MarkClientInterestDirty` when **only** the PC is received?

My reasoning was that getting ownership over any new actor could influence your interest. But thinking through it now, that was only when we were using things like the SpatialInterestComponent, which don't influence this, so I think we can do that. Will experiment.